src/CSI.IBTA.UserService/Services/UsersService.cs

```diff
@@ -0,0 +1,224 @@
+﻿using CSI.IBTA.DataLayer.Interfaces;
+using CSI.IBTA.Shared.DTOs;
+using CSI.IBTA.Shared.DTOs.Errors;
+using CSI.IBTA.Shared.Entities;
+using CSI.IBTA.Shared.Utils;
+using CSI.IBTA.UserService.Interfaces;
+using System.Net;
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+
+namespace CSI.IBTA.UserService.Services
+{
+    internal class UsersService : IUsersService
```

Comment:
You don't need to verify httpCOntext.User permissions at UsersService if it is already done at UserController. 
So, my vote would be to verify roles and credentials in controllers